### PR TITLE
Add Mac OS tracy info to profiling docs

### PR DESCRIPTION
**Pull request by mweatherley**
## Explanation

I got kind of lost on this earlier (the Tracy docs are not very helpful) and required some assistance, so I thought it might be helpful to add this somewhere in the profiling docs. The way it's presently inserted is kind of awkward, but I don't know enough about the other operating systems to make similar sections for them, which I think would be helpful, since it's going to be different on each one.

### CODE REVIEW

**Review by BD103 — APPROVED** [review]
Looks good, though this should be moved to the Contributor's Guide once that is more finished. :)

**Review by janhohenheim — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@mweatherley I'll merge this once the merge conflicts and linting errors are resolved.

**Comment by BD103** [comment]
This is the markdownlint error message for reference, copied from CI:

```
/github/workspace/docs/profiling.md:45:1 MD033/no-inline-html Inline HTML [Element: details]
/github/workspace/docs/profiling.md:46:3 MD033/no-inline-html Inline HTML [Element: summary]
/github/workspace/docs/profiling.md:48:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
/github/workspace/docs/profiling.md:48:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
/github/workspace/docs/profiling.md:49:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
/github/workspace/docs/profiling.md:49:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
/github/workspace/docs/profiling.md:50:1 MD004/ul-style Unordered list style [Expected: dash; Actual: asterisk]
/github/workspace/docs/profiling.md:50:1 MD007/ul-indent Unordered list indentation [Expected: 0; Actual: 2]
/github/workspace/docs/profiling.md:50:43 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
/github/workspace/docs/profiling.md:51:1 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 4]
```

**Comment by mweatherley** [comment]
Yeah, the issue is just that it has to be redone completely, since the current formatting can't actually be done in markdown without inline HTML. I can try rewriting the section in a way that incorporates it, I guess.

File: docs/profiling.md
Comment (djeedai) [inline docs/profiling.md:45]: nit: it's "macOS" not any other capitalizing/spacing form.
```suggestion
On macOS, Tracy can be installed through Homebrew by running `brew install tracy`, and the GUI client can be launched by running `tracy`. 
```

File: docs/profiling.md
Comment (djeedai) [inline docs/profiling.md:51]:
```suggestion
Note that on Windows, the capture tool is called `capture.exe`. On macOS, it is `tracy-capture` — e.g. `tracy-capture -o my_capture.tracy`. 
```

**Review by djeedai — APPROVED** [review]

File: docs/profiling.md
Comment (janhohenheim) [inline docs/profiling.md:49]: mdlint is not liking the trailing whitespace:
```suggestion
On macOS, Tracy can be installed through Homebrew by running `brew install tracy`, and the GUI client can be launched by running `tracy`.

In one terminal, run:
`./capture-release -o my_capture.tracy`
This will sit and wait for a tracy-instrumented application to start, and when it does, it will automatically connect and start capturing.
```

**Review by janhohenheim — CHANGES_REQUESTED** [review]

**Pull request merged**
